### ISSUE TITLE
Steering doesn't work sometimes

### ISSUE THREAD

**Issue by jhasse** [BeamNG/remotecontrol#8]
Many users report that gas and brake work, but steering doesn't.

**Comment by jhasse** [comment]
Okay the reason it fails, is that `magnet` is a null vector when the device doesn't have a magnetic sensor. This results in `normH` being 0 [here](http://grepcode.com/file/repository.grepcode.com/java/ext/com.google.android/android/5.1.1_r1/android/hardware/SensorManager.java#986) and thus `getRotationMatrix` returning false. Thus `SensorManger.getOrientation` never gets called in this function: 

``` java
    // calculates orientation angles from accelerometer and magnetometer output
    public void calculateAccMagOrientation() {
        if (SensorManager.getRotationMatrix(rotationMatrix, null, accel, magnet)) {
            SensorManager.getOrientation(rotationMatrix, accMagOrientation);
        }
    }
```

See https://github.com/BeamNG/remotecontrol/blob/77294470c5dcd6fd14343b11768765fcaa891f13/Android/Udpsteering/app/src/main/java/com/beamng/udpsteering/MainActivity.java#L276

`MainActivity.java` is only using `accMagOrientation[1]` [here](https://github.com/BeamNG/remotecontrol/blob/77294470c5dcd6fd14343b11768765fcaa891f13/Android/Udpsteering/app/src/main/java/com/beamng/udpsteering/MainActivity.java#L340). `getOrientation` sets this to `Math.asin(-rotationMatrix[7])` [here](http://grepcode.com/file/repository.grepcode.com/java/ext/com.google.android/android/5.1.1_r1/android/hardware/SensorManager.java#1267). `rotationMatrix[7]` is set to

``` java
gravity[1] / Math.sqrt(gravity[0]*gravity[0]+gravity[1]*gravity[1]+gravity[2]*gravity[2])
```

[here](http://grepcode.com/file/repository.grepcode.com/java/ext/com.google.android/android/5.1.1_r1/android/hardware/SensorManager.java#1007). `gravity` is `accel` in `MainActivity.java`.

To sum it up, the magnet sensor information isn't used at all. The calls to `getRotationMatrix` and `getOrientation` are unnecessary.
